tests/integration/incremental_unique_id_test/test_incremental_unique_id.py

```diff
@@ -0,0 +1,270 @@
+from dbt.exceptions import ParsingException
+from tests.integration.base import DBTIntegrationTest, use_profile
+from collections import namedtuple
+from pathlib import Path
+
+
+TestResults = namedtuple(
+    'TestResults',
+    ['seed_count', 'model_count', 'seed_rows', 'inc_test_model_count',
+     'opt_model_count', 'relation'],
+)
+
+
+class TestIncrementalUniqueKey(DBTIntegrationTest):
+    @property
+    def schema(self):
+        return 'incremental_unique_key'
+
+    @property
+    def models(self):
+        return 'models'
+
+    def count_test_relations(self, seed, incremental_model):
```

Comment:
Should we name this to something like create seed or create incremental models? count doesn't cover everything this function does. I feel like we can also just put this in the setup with some comment since this function is only used in one place

Reply:
Hey Chenyu, I went back and forth on this myself. Here's my thought.

In the abstraction of the code itself, the primary thing occurring is obtaining the number of models built (a number returned is a sign we are successful in building our code). The state of the models on the database itself is actually a "side effect" in my opinion. The returned model count is a number only obtained because I was able to successfully create the database. Hence, the function should be named in a manner that reflects the count of test relations (seeds and tables) built.

Do you buy that? Got a better name in mind? Contrary thoughts? :)

Reply:
In my mind, the count being returned is more like a side effect. The function you call in this function is actually `run_dbt` which is named based on the actual thing that happened. But you did a `len()` operation on the result returned. So I guess you can go both ways here. 
But I would avoid abstracting this and the bottom one being a function since they are not being used in multiple places, there are more things going on here than just the function name indicates, and reading the code itself is not hard. `setup_test` is a perfect name for all of the things happening